Content.Server/StationRecords/StationRecordSet.cs

```diff
@@ -52,43 +54,43 @@ public IEnumerable<(StationRecordKey, T)> GetRecordsOfType<T>()
     }
 
     /// <summary>
-    ///     Add an entry into a record.
+    ///     Create a new record with an entry.
     /// </summary>
     /// <param name="entry">Entry to add.</param>
     /// <typeparam name="T">Type of the entry that's being added.</typeparam>
-    public StationRecordKey AddRecordEntry<T>(EntityUid station, T entry)
+    public uint? AddRecordEntry<T>(T entry)
     {
         if (entry == null)
-            return StationRecordKey.Invalid;
+            return null;
 
-        var key = new StationRecordKey(_currentRecordId++, station);
+        var key = _currentRecordId++;
         AddRecordEntry(key, entry);
         return key;
     }
 
     /// <summary>
-    ///     Add an entry into a record.
+    ///     Add an entry into an existing record.
     /// </summary>
     /// <param name="key">Key for the record.</param>
     /// <param name="entry">Entry to add.</param>
     /// <typeparam name="T">Type of the entry that's being added.</typeparam>
-    public void AddRecordEntry<T>(StationRecordKey key, T entry)
+    public void AddRecordEntry<T>(uint key, T entry)
     {
         if (entry == null)
             return;
 
-        if (Keys.Add(key))
-            _tables.GetOrNew(typeof(T))[key] = entry;
+        Keys.Add(key);
+        _tables.GetOrNew(typeof(T))[key] = entry;
```

Comment:
its needed since otherwise trying to add a criminal record to someones station record would do nothing as *a* record already exists, even if it isnt the same type